httpcore5/src/main/java/org/apache/hc/core5/reactor/AbstractIOSessionPool.java

```diff
@@ -183,7 +183,20 @@ private void getSessionInternal(
             } else {
                 poolEntry.requestQueue.add(callback);
                 if (poolEntry.sessionFuture != null && poolEntry.sessionFuture.isDone()) {
-                    poolEntry.sessionFuture = null;
+                    // Check whether we should recreate a new conn or not
+                    try {
+                        poolEntry.session = poolEntry.sessionFuture.get();
```

Comment:
@gaojieliu This is wrong. One _MUST NOT_ use a blocking call inside a callback that can be executed by an I/O reactor thread. 
There has to be a better way.